### PR TITLE
Add RHOSP network template automation

### PR DESCRIPTION
**Pull request by wangxian-ibm**
- writes network config templates 
- supports Redhat/CentOS and Ubuntu

### CODE REVIEW

File: roles/common/tasks/networking.yml [hunk 1]
```diff
@@ -1,16 +1,67 @@
...
-- name: network interfaces file
-  copy: content="{{ network_interfaces }}" dest=/etc/network/interfaces
-        force=yes owner=root group=root mode=0644
-  when: network_interfaces is defined
-  tags: ['prereboot']
+- block:
+  - name: set network config dir path - ubuntu
+    set_fact:
+      network_dir_path: /etc/network/interfaces.d
+    when: os == 'ubuntu'
```
Comment (epwn) [inline roles/common/tasks/networking.yml]: can we refactor the os checks to use ` ansible_os_family`?

File: roles/common/tasks/networking.yml [hunk 2]
```diff
@@ -1,16 +1,67 @@
 ---
-- name: network interfaces file
-  copy: content="{{ network_interfaces }}" dest=/etc/network/interfaces
-        force=yes owner=root group=root mode=0644
-  when: network_interfaces is defined
-  tags: ['prereboot']
+- block:
```
Comment (paulczar) [inline roles/common/tasks/networking.yml]: is this block actually doing anything?

**Comment by paulczar** [comment]
ok to test

**Comment by scoop206** [comment]
:+1:

File: roles/common/tasks/networking.yml (hunk 1 above)
Comment (wangxian-ibm) [inline roles/common/tasks/networking.yml]: This is tied to this card. https://trello.com/c/Mrv3ZTGc/87-use-ansible-os-family-instead-of-os
This should be fixed when a solution is decided for this task.

**Pull request merged**